### PR TITLE
parameter & operation extensions display

### LINKED ISSUE #3052: X-attributes can not be displayed in the Parameters or Operation section in swagger ui.
## issue
X-attributes can not be displayed in the Parameters section in swagger ui.
But, it can display in the property of model.
I think X-attributes should  be displayed in the Parameters section.

## version
 swagger-ui version; master branch(4/19/2017)

## swagger file example
```
swagger: '2.0'
info:
  version: 1.0.0
  title: Swagger Petstore
basePath: /shop
consumes:
  - application/json
produces:
  - application/json
paths:
  /pets:
    get:
      summary: get 
      parameters:
        - name: petName
          in: query
          type: string
          x-domain: domainName ## This can not be displayed
      responses:
        '200':
          description: list
          schema:
            type: array
            items:
              $ref: '#/definitions/ResPet'
definitions:
  ResPet:
    type: object
    required:
      - id
      - name
    properties:
      id:
        example: 123456
        type: integer
        format: int64
      name:
        example: taro
        type: string
        x-domain: domainName ## This can be displayed.
```

### PR DESCRIPTION
**Pull request by thompsongl**
Fixes #3052 by adding components to display Parameter and Operation extensions (`x-` values)

Presentational components for `x-` value displays are small and explicit intentionally. These changes allow for baseline rendering while giving easy access to override/wrap each extension component granularly. For instance, we plan on wrapping operation extensions to display extra information about our security-related values. 

_Considerations:_
* Feedback on location, layout, and style are appreciated. This is an initial pass and welcome insight into Swagger design & UX choices/patterns.
* Same feedback appreciated on component naming conventions.

### CODE REVIEW

**Comment by webron** [comment]
@thompsongl thanks for the PR. I'll leave it to @shockey to comment about the code itself. Running this with a definition that has an extension in a parameter - am I supposed to see anything? Could be completely blind...

**Comment by thompsongl** [comment]
@webron Not blind. I had an uncommitted change hanging out 😳
You should now see extension defs under the `type`, `in`, etc., values

**Comment by webron** [comment]
I think for an initial version, the styling is ok. Not a lot of magic that can be done with it.

However, I think rendering the extensions should be optional and by default should be false.

**Comment by thompsongl** [comment]
I can make that happen. Optional by way of configuration parameters or another method?
Model extensions are currently displayed by default. Would you like to see those hidden by default as well or are they fine as is?

**Comment by webron** [comment]
Yeah, a configuration parameter. And yeah, would be great if that would apply to model extensions as well. Thanks for the help.

**Comment by thompsongl** [comment]
@shockey #3875 is (somewhat) blocking this. Specifically, I made `getConfigs` available inside of `<ParameterRow />` there, which we'll now need to read the extensions flag.
Just an FYI that there's overlap. I can adjust commits if #3875 can't merge for whatever reason

**Comment by shockey** [comment]
@thompsongl, I just merged #3875 😄

**Comment by thompsongl** [comment]
Updated: Uses configuration param to display extensions
Most of the diff is now related to getting `getConfigs` where needed.
Also, note that (as requested) model extensions are no longer displayed by default. Breaking change in a sense; at the very least worth detailing in release notes

**Comment by thompsongl** [comment]
@shockey, anything else you'd like to see as part of this changeset?

**Comment by shockey** [comment]
@thompsongl, all good here - hoping to get this reviewed and merged Monday 😄

File: src/core/components/array-model.jsx [hunk 1]
```diff
@@ -7,6 +7,7 @@ export default class ArrayModel extends Component {
   static propTypes = {
     schema: PropTypes.object.isRequired,
     getComponent: PropTypes.func.isRequired,
+    getConfigs: PropTypes.func.isRequired,
```
Comment (shockey) [inline src/core/components/array-model.jsx]: I'm not a huge fan of the `getConfigs` handaround - would prefer to see a container component mapping from the configs to props for children.

**Pull request merged**